src/main/java/house/inksoftware/systemtest/domain/config/infra/mock/SystemTestMockedServerLauncher.java

```diff
@@ -1,64 +1,126 @@
+import com.fasterxml.jackson.databind.node.ObjectNode;
 import house.inksoftware.systemtest.domain.config.infra.SystemTestResourceLauncher;
 import house.inksoftware.systemtest.domain.config.infra.db.postgres.SystemTestPostgresLauncher;
 import lombok.RequiredArgsConstructor;
 import lombok.SneakyThrows;
 import lombok.extern.slf4j.Slf4j;
-import org.junit.Rule;
+import org.awaitility.core.ConditionTimeoutException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.testcontainers.containers.BindMode;
-import org.testcontainers.containers.Container;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
 import org.testcontainers.containers.GenericContainer;
-import org.testcontainers.containers.PostgreSQLContainer;
-import org.testcontainers.utility.DockerImageName;
 import org.testcontainers.utility.MountableFile;
 
-import javax.sql.DataSource;
 import java.io.File;
+import java.io.IOException;
+import java.net.URI;
 import java.nio.file.Files;
+import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.util.*;
-import java.util.stream.Collectors;
+import java.nio.file.StandardOpenOption;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.awaitility.Awaitility.await;
 
 @Slf4j
 @RequiredArgsConstructor
 public class SystemTestMockedServerLauncher implements SystemTestResourceLauncher {
     private static final Logger LOGGER = LoggerFactory.getLogger(SystemTestPostgresLauncher.class);
 
+    private final TestRestTemplate restTemplate;
     private final String relativePathToConfig;
     private final Integer warmupSeconds;
     private GenericContainer container;
 
 
-
     @SneakyThrows
     @Override
     public void setup() {
         if (container == null) {
-
-            String fileName = relativePathToConfig.substring(relativePathToConfig.lastIndexOf("/"));
+            File updatedConfigFile = addStatusVerificationEndpoint(relativePathToConfig);
 
             Map<String, String> envVariables = new HashMap<>();
-            envVariables.put("MOCKSERVER_INITIALIZATION_JSON_PATH", "/config/" + fileName);
+            envVariables.put("MOCKSERVER_INITIALIZATION_JSON_PATH", "/config/" + updatedConfigFile.getName());
 
             container = new GenericContainer("mockserver/mockserver")
-                    .withCopyFileToContainer(MountableFile.forClasspathResource(relativePathToConfig), "/config/" + fileName)
+                    .withCopyFileToContainer(MountableFile.forHostPath(updatedConfigFile.getPath()), "/config/" + updatedConfigFile.getName())
                     .withEnv(envVariables);
 
             container.setPortBindings(Arrays.asList("1080:1080/tcp"));
+            container.withReuse(true);
             container.start();
             Thread.sleep(warmupSeconds * 1000);
```

Comment:
why sleep if you'll waitUntilEndpointsAreAvailable?

Reply:
ah, it was for local testing only :D